pkg/build/graph/analysis/bc.go

```diff
@@ -162,6 +162,62 @@ func FindCircularBuilds(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
 	return markers
 }
 
+// findPendingTags is the guts behind FindPendingTags .... break out some of the content and reduce some indentation
+func findPendingTags(uncastIstNode graph.Node, g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+	istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode)
+	if istNode.Found() {
+		return markers
+	}
+	bcNodes := buildedges.BuildConfigsForTag(g, uncastIstNode)
+	for _, bcNode := range bcNodes {
+		latestBuild := buildedges.GetLatestBuild(g, bcNode)
+
+		// A build config points to the non existent tag but no current build exists.
+		if latestBuild == nil {
+			markers = append(markers, osgraph.Marker{
+				Node:         graph.Node(bcNode),
+				RelatedNodes: []graph.Node{uncastIstNode},
+
+				Severity:   osgraph.WarningSeverity,
+				Key:        TagNotAvailableWarning,
+				Message:    fmt.Sprintf("%s needs to be imported or created by a build.", f.ResourceName(istNode)),
```

Comment:
We should print this only once and not for every build config. So if there is no build by our build configs, then append this marker (should probably happen after this loop). Also the suggestion below should be different on multiple build configs?